dfastmi/batch/ConfigurationChecker.py

```diff
@@ -27,402 +27,92 @@
 This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
 """
 
-from typing import Optional, Tuple
-from dfastmi.io.Reach import ReachLegacy
-from dfastmi.io.RiversObject import RiversObject
-from dfastmi.kernel.core import Vector, BoolVector, QRuns
-from dfastmi.io.ApplicationSettingsHelper import ApplicationSettingsHelper
-
-import dfastmi.kernel.core
-import dfastmi.plotting
 import configparser
+from typing import Tuple
+from dfastmi.batch.AConfigurationChecker import AConfigurationCheckerBase
+from dfastmi.io.RiversObject import RiversObject
 
-WAQUA_EXPORT = "WAQUA export"
-DFLOWFM_MAP = "D-Flow FM map"
-
-def check_configuration_v1(rivers: RiversObject, config: configparser.ConfigParser) -> bool:
-    """
-    Check if a version 1 analysis configuration is valid.
-
-    Arguments
-    ---------
-    rivers: RiversObject
-        A dictionary containing the river data.
-    config : configparser.ConfigParser
-        Configuration for the D-FAST Morphological Impact analysis.
-
-    Returns
-    -------
-    success : bool
-        Boolean indicating whether the D-FAST MI analysis configuration is valid.
-    """
-    try:
-        reach = get_reach(rivers, config)
-    except:
-        return False
-
-    nwidth = reach.normal_width
-    [_, apply_q, _, _, _, _, _, _] = get_levels_v1(reach, config, nwidth)
-    
-    mode_str = config["General"].get("Mode", DFLOWFM_MAP)
-    for i in range(3):
-        if not apply_q[i]:
-            continue
-        
-        if not check_configuration_v1_cond(config, mode_str, i):
+class ConfigurationChecker(AConfigurationCheckerBase):
+    """
+        Check if a version 2 analysis configuration is valid.
+    """
+    def check_configuration(self,
+                            rivers: RiversObject,
+                            config: configparser.ConfigParser) -> bool:
+        """
+        Check if a version 2 analysis configuration is valid.
+
+        Arguments
+        ---------
+        rivers: RiversObject
+            A dictionary containing the river data.
```

Comment:
is RiversObject a dictionary containing river data?

Reply:
updated